lib/registration.pm

```diff
@@ -57,18 +57,25 @@ our %SLE15_DEFAULT_MODULES = (
 sub fill_in_registration_data {
     my ($addon, $uc_addon);
     if (!get_var("HDD_SCC_REGISTERED")) {
-        # SLE 15 & textmode
-        if (sle_version_at_least('15') && check_var('DESKTOP', 'textmode')) {
-            send_key "alt-m";    # select email field if yast2 add-on
+        if (!get_var("SMT_URL")) {
+            # SLE 15 & textmode
+            if (sle_version_at_least('15') && check_var('DESKTOP', 'textmode')) {
+                send_key "alt-m";    # select email field if yast2 add-on
+            }
+            else {
+                send_key "alt-e";    # select email field if installation
+            }
+            send_key "backspace";    # delete m or e
+            type_string get_required_var('SCC_EMAIL') if get_var('SCC_EMAIL');
+            save_screenshot;         # show typed value or empty fields also as synchronization
+            send_key "alt-c";        # select registration code field
+            type_string get_required_var('SCC_REGCODE') if get_var('SCC_REGCODE');
```

Comment:
That method is getting too complex, please define some functions and extract the code

Reply:
How about add new functions such as type_scc_email and type_scc_regcode?

Reply:
maybe you understood me wrong. The method "fill_in_registration_data" is getting too complex. The indendation level is becoming too deep. Better extract a function for the whole if/else block starting in line L59

Reply:
extracted this if/else block to "fill_in_reg_server" function.